src/Evolution/DgSubcell/Tags/Jacobians.hpp

```diff
@@ -35,4 +35,76 @@ struct DetInverseJacobianLogicalToGrid : db::SimpleTag {
   static std::string name() { return "Det(InverseJacobian(Logical,Grid))"; }
   using type = std::optional<Scalar<DataVector>>;
 };
+
+/// \brief The inverse Jacobian from the element logical frame to the inertial
+/// frame at the cell centers.
+///
+/// Specifically, \f$\partial x^{\bar{i}} / \partial x^i\f$, where \f$\bar{i}\f$
+/// denotes the element logical frame and \f$i\f$ denotes the inertial frame.
+///
+/// \note stored as a `std::optional` so we can reset it when switching to DG
+/// and reduce the memory footprint (is it useful for a ComputeTag though?).
+template <size_t Dim>
+struct InverseJacobianLogicalToInertial : db::SimpleTag {
+  static std::string name() { return "InverseJacobian(Logical,Inertial)"; }
+  using type =
+      std::optional<::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                                      Frame::Inertial>>;
+};
+
+/// Compute item for the inverse jacobian matrix from logical to
+/// inertial coordinates
+template <typename MapTagGridToInertial, size_t Dim>
+struct InverseJacobianLogicalToInertialCompute
+    : InverseJacobianLogicalToInertial<Dim>,
+      db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = InverseJacobianLogicalToInertial<Dim>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<MapTagGridToInertial,
+                 evolution::dg::subcell::Tags::Coordinates<dim, Frame::Grid>,
+                 ::Tags::Time, ::domain::Tags::FunctionsOfTime,
+                 InverseJacobianLogicalToGrid<Dim>>;
+  static void function(
+      const gsl::not_null<return_type*> inverse_jacobian_logical_to_inertial,
+      const ::domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, dim>&
+          grid_to_inertial_map,
+      const tnsr::I<DataVector, dim, Frame::Grid>& grid_coords,
+      const double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>&
+          functions_of_time,
+      std::optional<::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                                      Frame::Grid>>
+          inverse_jacobian_logical_to_grid) {
+    if (inverse_jacobian_logical_to_grid.has_value()) {
```

Comment:
I ran into the issue that the logical->grid Jacobian is null at t=0 before taking the first time step, so the initial observation errored with "bad optional access".

Reply:
Maybe a discussion for the code meeting later; I'm not sure what the solution would be here.

Reply:
This is because it's currently only being computed in https://github.com/sxs-collaboration/spectre/blob/b1451b1e87807b37726484b69a656dd93c4cc6ab/src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp#L81 We could add a compute tag for it. I think that should fix the issue and guarantee that the value is always valid (and remove the `std::optional`).

Reply:
Do you mean a compute tag for the logical-to-grid inverse jacobian?
[And then removing std::optional for both the logical-to-grid and logical-to-inertial inverse jacobians?]

Reply:
Yea, that's right! And then the l2i can just set the DataVector pointers to the l2g one when the mesh is static. That makes the computation super cheap :D